wire-ios-sync-engine/Source/Synchronization/Strategies/EvaluateOneOnOneConversationsStrategy.swift

```diff
@@ -36,15 +40,52 @@ final class EvaluateOneOnOneConversationsStrategy: AbstractRequestStrategy {
         super.init(withManagedObjectContext: managedObjectContext, applicationStatus: applicationStatus)
     }
 
+    deinit {
+        task?.cancel()
+    }
+
     override func nextRequest(for apiVersion: APIVersion) -> ZMTransportRequest? {
-        if isSyncing {
-            // TODO: [WPB-5812] epic implementation
-            // - Produce a list of the users we have 1:1 conversations with
-            // - Loop through the list and evaluate the 1:1 conversation for each user.
+        guard isSyncing, task == nil else {
+            return nil
+        }
+
+        WireLogger.conversation.info("EvaluateOneOnOneConversationsStrategy: start evaluate one on one conversations!")
+
+        // store task to avoid duplicated entries and concurrency issues
+        task = Task { [weak self] in
+            guard let self, let syncContext = managedObjectContext.performAndWait({ self.managedObjectContext.zm_sync }) else {
```

Comment:
what is the value of managedObjectContext? syncContext already, no?

Reply:
this is an implicit assumption, that was not clear to me.. did some change but added `precondition` to make it better understandable for the reader.